genesyscloud/outbound_ruleset/genesyscloud_outbound_ruleset_proxy.go

```diff
@@ -131,10 +133,15 @@ func getAllOutboundRulesetFn(ctx context.Context, p *outboundRulesetProxy) (*[]p
 func getOutboundRulesetByIdFn(ctx context.Context, p *outboundRulesetProxy, rulesetId string) (ruleset *platformclientv2.Ruleset, statusCode int, err error) {
 	ruleset, resp, err := p.outboundApi.GetOutboundRuleset(rulesetId)
 	if err != nil {
-		return nil, resp.StatusCode, fmt.Errorf("Failed to ruleset by id %s: %s", rulesetId, err)
+		//This is an API that throws an error on a 404 instead of just returning a 404.
+		if strings.Contains(fmt.Sprintf("%s", err), "API Error: 404") {
```

Comment:
I think this is the cause of the failing test where we have a dangling pointer.  @charliecon @dginty4 @HemanthDogiparthi12 We need to really aggressive that our tests are passing before we merge a PR